jwst/assign_wcs/util.py

```diff
@@ -562,8 +563,17 @@ def create_grism_bbox(input_model,
     instr_name = input_model.meta.instrument.name
     if instr_name == "NIRCAM":
         filter_name = input_model.meta.instrument.filter
+        if input_model.meta.instrument.pupil.endswith('R'):
+            # _extraction_height_direction is used if extract_height is not None
+            _extract_height_direction = 'y'
+        else:
+            _extract_height_direction = 'x'
```

Comment:
The DISPAXIS or `meta.wcsinfo.dispersion_direction` keyword should give the dispersion direction without having to rely on filter name parsing.  Or move the filter name parsing to the dispersion direction function

https://github.com/spacetelescope/jwst/blob/ff9ea4017ebdc62309cf97661a5093f72c402722/jwst/lib/dispaxis.py#L6-L8